File: src/autofill/CredentialProviderViewController.m
//
//  CredentialProviderViewController.m
//  lol
//
//  Created by Sebastian Livoni on 31/01/2025.
//

#import "CredentialProviderViewController.h"

@implementation CredentialProviderViewController

- (void)prepareCredentialListForServiceIdentifiers:
    (NSArray<ASCredentialServiceIdentifier *> *)serviceIdentifiers {
}

- (void)provideCredentialWithoutUserInteractionForRequest:
    (id<ASCredentialRequest>)credentialRequest {
  ASPasswordCredential *passwordCredential =
      [[ASPasswordCredential alloc] initWithUser:@"username"
                                        password:@"password"];
  [self.extensionContext
      completeRequestWithSelectedCredential:passwordCredential
                          completionHandler:nil];
}

- (void)exitWithUserInteractionRequired {
    [self.extensionContext cancelRequestWithError:[NSError errorWithDomain:ASExtensionErrorDomain
                                                                      code:ASExtensionErrorCodeUserInteractionRequired
                                                                  userInfo:nil]];
}

@end

File: src/macos_autofill/CredentialProviderViewController.swift
//
//  CredentialProviderViewController.h
//  CredentialProviderViewController
//
//  Created by Sebastian Livoni on 27/01/2025.
//

import AuthenticationServices

class CredentialProviderViewController: ASCredentialProviderViewController {

    override func prepareCredentialList(for serviceIdentifiers: [ASCredentialServiceIdentifier]) {
    }

    override func provideCredentialWithoutUserInteraction(
        for credentialRequest: any ASCredentialRequest
    ) {
        let passwordCredential = ASPasswordCredential(user: "username", password: "password")

        extensionContext.completeRequest(
            withSelectedCredential: passwordCredential, completionHandler: nil)
    }
}
